src/node_i18n.cc

```diff
@@ -602,6 +604,37 @@ void SetDefaultTimeZone(const char* tzid) {
   CHECK(U_SUCCESS(status));
 }
 
+void SetDefaultLocale(const char* localeid) {
+  UErrorCode status = U_ZERO_ERROR;
+
+  // Set the locale to the requested locale, no matter if it is
+  // supported or not. Let ICU handle the conversion to a supported locale.
+  // E.g. "en_US" and "en-US" will be internally converted to "en_US".
+  uloc_setDefault(localeid, &status);
+  CHECK(U_SUCCESS(status));
+
+  // Now check if the locale was actually set to a supported locale.
+  // We iterate over all supported locales and check if the default locale
+  // is among them. If so, we can finish here.
+  const char* newDefaultLocale = uloc_getDefault();
+  int newDefaultLocaleLen = strlen(newDefaultLocale);
+  int32_t locCount = 0;
+  const icu::Locale* supportedLocales =
+      icu::Locale::getAvailableLocales(locCount);
+  for (int32_t i = 0; i < locCount; ++i) {
+    if (strncmp(newDefaultLocale,
+                supportedLocales[i].getName(),
+                newDefaultLocaleLen) == 0) {
+      return;
+    }
+  }
```

Comment:
I'm not sure of the purpose of this. Just let the user set the default locale to whatever they want.  The user's own system might be set to something not available.

Reply:
You can set the --icu-locale to invalid values and as long as it is in the right format, it will set it. It will only go to und if it is an invalid locale.This ensures, that the locale is existing and valid. 

Tbh. I didnt know how I could get the BestMatcher Class running. So I wrote this logic.

Reply:
what's the purpose of trying to protect the user from invalid values?

Reply:
See comment by @bnoordhuis 

https://github.com/nodejs/node/pull/50538#discussion_r1382441550